x/subspaces/keeper/msg_server_feegrant.go

```diff
@@ -0,0 +1,148 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/desmos-labs/desmos/v4/x/subspaces/types"
+)
+
+// GrantUserAllowance defines a rpc method for MsgGrantUserAllowance
+func (k msgServer) GrantUserAllowance(goCtx context.Context, msg *types.MsgGrantUserAllowance) (*types.MsgGrantUserAllowanceResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if !k.HasSubspace(ctx, msg.SubspaceID) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "subspace with id %d not found", msg.SubspaceID)
+	}
+	if !k.HasPermission(ctx, msg.SubspaceID, types.RootSectionID, msg.Granter, types.PermissionGrantAllowances) {
+		return nil, sdkerrors.Wrap(types.ErrPermissionDenied, "you cannot grant allowances in this subspace")
+	}
+	if k.HasUserGrant(ctx, msg.SubspaceID, msg.Granter, msg.Grantee) {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "fee allowance already exists")
+	}
+
+	allowance, err := msg.GetUnpackedAllowance()
+	if err != nil {
+		return nil, err
+	}
+	grant, err := types.NewUserGrant(msg.SubspaceID, msg.Granter, msg.Grantee, allowance)
+	if err != nil {
+		return nil, err
+	}
+	k.Keeper.SaveUserGrant(ctx, grant)
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+			sdk.NewAttribute(sdk.AttributeKeyAction, sdk.MsgTypeURL(msg)),
+			sdk.NewAttribute(sdk.AttributeKeySender, msg.Granter),
+		),
+		sdk.NewEvent(
+			types.EventTypeGrantUserAllowance,
+			sdk.NewAttribute(types.AttributeKeySubspaceID, fmt.Sprintf("%d", msg.SubspaceID)),
+			sdk.NewAttribute(types.AttributeKeyGranter, msg.Granter),
+			sdk.NewAttribute(types.AttributeKeyGrantee, msg.Grantee),
+		),
+	})
+	return &types.MsgGrantUserAllowanceResponse{}, nil
+}
+
+// RevokeUserAllowance defines a rpc method for MsgRevokeUserAllowance
+func (k msgServer) RevokeUserAllowance(goCtx context.Context, msg *types.MsgRevokeUserAllowance) (*types.MsgRevokeUserAllowanceResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if !k.HasUserGrant(ctx, msg.SubspaceID, msg.Granter, msg.Grantee) {
```

Comment:
I think we should also check the permission to manage allowances here

Reply:
I finally decided not to check the permission in the `Revoke` since it makes the granter could not remove the grants granted when he had the permission.